### DISCUSSION TITLE
Http\Concerns\InteractsWithContentType expectJson check fails for multiple accept options

### DISCUSSION THREAD

**Discussion by muisit** [laravel/framework#33389] [General] [answered]
I'm trying to build a GraphQL client interface using Apollo-GraphQL and rebing/graphql-laravel. I am running into an issue where I (by accident) send the Accept header containing multiple options: "Accept: */*, application/json"
By default, Apollo-GraphQL sends the "Accept: */*" header and when I manually configure an override using "Accept: application/json", there is a mismatch in the header key names (Accept vs accept), which causes Apollo-link-http to include both headers and makes the underlying Fetch API merge the headers.

Anyway: "Accept: */*, application/json" is by itself a valid, although slightly confusing header. A less confusing, but valid header would be "Accept: text/html, application/json", which will trigger this issue as well.

Http\Concerns\InteractsWithContentType only checks the first entry in the accepted headers list. In order to get your Laravel back-end to recognise that this response expects JSON (or could expect JSON), you need to force the */json content type as the first content type.

Expected behaviour:
InteractsWithContentType::expectsJson should loop over all the entries in the Accept header and test each entry, returning true if any of them matches the json check. In that case, the request expects a lot, but also JSON.

**Comment by taylorotwell** [comment, marked as answer]
`expectsJson` specifically checks if `application/json` is the FIRST requested content-type. However, the `acceptsJson` method sounds more like what you want. It determines if JSON is accepted at all.
